module/VuFind/src/VuFind/View/Helper/Root/SchemaOrg.php

```diff
@@ -106,4 +111,53 @@ public function getMeta(string $property, string $content, array $attributes = [
     {
         return $this->getTag('meta', compact('property', 'content') + $attributes);
     }
+
+    /**
+     * Get default record types as array.
+     *
+     * @return array
+     */
+    public function getDefaultRecordTypesArray(): array
+    {
+        if ($this->ils->__invoke()->getOfflineMode() !== 'ils-none') {
+            return ['Product'];
+        }
+        return [];
+    }
+
+    /**
+     * Get default record types as string.
+     *
+     * @return string
+     */
+    public function getDefaultRecordTypes(): string
+    {
+        return implode(' ', $this->getDefaultRecordTypesArray());
+    }
+
+    /**
+     * Get all record types for the given record.
+     *
+     * @param DefaultRecord $driver Record Driver
+     *
+     * @return array
+     */
+    public function getRecordTypesArray(DefaultRecord $driver): array
+    {
+        $defaultTypes = $this->getDefaultRecordTypesArray();
+        $recordTypes = $driver->getSchemaOrgFormatsArray();
+        return array_merge($defaultTypes, $recordTypes);
```

Comment:
Maybe it's worth adding an `array_unique` around this to avoid duplication (just to be on the safe side).